### PR TITLE
Add an inference notebook based on custom bundle

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**

### CODE REVIEW

**Comment by yiheng-wang-nv** [comment]
> Hi @yiheng-wang-nv , I have some markdown changes in #78 . Can we merge that one first, and then also modify the notebook "Perform real-time inference" with the multipart encoder? Thanks!

Sure, will change after your PR is merged

File: notebooks/Perform Real-time Inference.ipynb [hunk 1]
```diff
@@ -217,18 +221,32 @@
...
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`MultipartDecoder` is used to decode the response data. If it's not installed, you can use the following command to install it:\n",
+    "\n",
+    "```Bash\n",
+    "pip install requests_toolbelt==1.0.0\n",
```
Comment (mingxin-zheng) [inline notebooks/Perform Real-time Inference.ipynb:231]: Please add a section on the top: "Environment Setup" like other MONAI tutorials

File: notebooks/Perform Real-time Inference.ipynb [hunk 2]
```diff
@@ -217,18 +221,32 @@
...
+  },
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": [
-    "attachment_data = response.content\n",
-    "tmp_save_path = \"pred.nrrd\"\n",
+    "from requests_toolbelt.multipart.decoder import MultipartDecoder\n",
```
Comment (mingxin-zheng) [inline notebooks/Perform Real-time Inference.ipynb:241]: Please move this to the import cell on the top.

File: notebooks/Perform Real-time Inference with a Custom MONAI Bundle.ipynb [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from requests_toolbelt.multipart.decoder import MultipartDecoder\n",
```
Comment (mingxin-zheng) [inline notebooks/Perform Real-time Inference with a Custom MONAI Bundle.ipynb:352]: Similarly, please move this to the import cell on the top

File: notebooks/Perform Real-time Inference with a Custom MONAI Bundle.ipynb [hunk 4]
```diff
@@ -0,0 +1,450 @@
...
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`MultipartDecoder` is used to decode the response data. If it's not installed, you can use the following command to install it:\n",
+    "\n",
+    "```Bash\n",
+    "pip install requests_toolbelt==1.0.0\n",
```
Comment (mingxin-zheng) [inline notebooks/Perform Real-time Inference with a Custom MONAI Bundle.ipynb:342]: Similarly, please create a "Environment Setup" cell on the top.

File: notebooks/Perform Real-time Inference with a Custom MONAI Bundle.ipynb [hunk 5]
```diff
@@ -0,0 +1,450 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Guide to do Real-time Inference on a Custom MONAI Bundle with NVIDIA Cloud APIs\n",
```
Comment (mingxin-zheng) [inline notebooks/Perform Real-time Inference with a Custom MONAI Bundle.ipynb:8]: Guide to do Real-time Inference on a Custom MONAI Bundle with NVIDIA Cloud APIs

Guide to do Real-Time Inference Using a Custom MONAI Bundle with NVIDIA Cloud APIs

File: notebooks/Perform Real-time Inference with a Custom MONAI Bundle.ipynb [hunk 6]
```diff
@@ -0,0 +1,450 @@
...
+    "\n",
+    "In this guide, we will guide you through the process of setting up a real-time inference system on a custom MONAI bundle with MONAI cloud APIs.We will cover setting up the experiments, making on-the-fly predictions, and managing the outputs to ensure a seamless, efficient, and real-time decision-making pipeline.\n",
```
Comment (mingxin-zheng) [inline notebooks/Perform Real-time Inference with a Custom MONAI Bundle.ipynb:10]: Can be improved as

```
In this guide, we will walk you through setting up a real-time inference system using a custom MONAI bundle with MONAI Cloud APIs. We will cover experiment setup, making predictions on-the-fly, and managing outputs to ensure a seamless, efficient, and real-time decision-making pipeline.
```

File: notebooks/Perform Real-time Inference with a Custom MONAI Bundle.ipynb [hunk 7]
```diff
@@ -0,0 +1,450 @@
...
+    "\n",
+    "## Table of Contents\n",
+    "\n",
+    "- Introduction\n",
+    "- Setup\n",
+    "- Dataset Creation\n",
+    "- Custom MONAI Bundle Creation\n",
+    "- Configuring Experiment to enable the real-time inference\n",
```
Comment (mingxin-zheng) [inline notebooks/Perform Real-time Inference with a Custom MONAI Bundle.ipynb:18]: - Configuring Experiment to enable the real-time inference
- Prepare the image ID for the inference request
- Cleaning up

Should be in the title format, for example:
Configuring Experiment to Enable the Real-Time Inference

File: notebooks/Perform Real-time Inference with a Custom MONAI Bundle.ipynb [hunk 8]
```diff
@@ -0,0 +1,450 @@
...
+   "metadata": {
+    "tags": [
+     "parameters"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# API Endpoint and Credentials\n",
+    "host_url = \"https://api.monai.ngc.nvidia.com\"\n",
+    "ngc_api_key = os.environ.get('MONAI_API_KEY')  # we recommend using environment variables for API keys, but you can also hardcode them here\n",
```
Comment (mingxin-zheng) [inline notebooks/Perform Real-time Inference with a Custom MONAI Bundle.ipynb:62]: Can we change this to

```
ngc_api_key = os.environ.get("MONAI_API_KEY", "<YOUR_API_KEY>")  # we recommend using environment variables for API keys, but you can also hardcode them here
```

**Review by mingxin-zheng — APPROVED** [review]
Hi @yiheng-wang-nv , please fix the minor suggestions above and then merge it once you resolve all the conversations. Thanks!

**Pull request merged**